airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/BigQueryDestinationTestUtils.java

```diff
@@ -0,0 +1,133 @@
+   * @param configFile
+   * @param datasetId
+   * @return
+   * @throws IOException
+   */
+  public static JsonNode createConfig(Path configFile, String datasetId) throws IOException {
+    final String tmpConfigAsString = Files.readString(configFile);
+    final JsonNode tmpConfigJson = Jsons.deserialize(tmpConfigAsString);
+    return Jsons.jsonNode(((ObjectNode)tmpConfigJson).put(BigQueryConsts.CONFIG_DATASET_ID, datasetId));
+  }
+
+  /**
+   * Get a handle for the BigQuery dataset instance used by the test.
+   * This dataset instance will be used to verify results of test operations
+   * and for cleaning up after the test runs
+   * @param config
+   * @param bigquery
+   * @param datasetId
+   * @return
+   */
+  public static Dataset initDataSet(JsonNode config, BigQuery bigquery, String datasetId) {
+    final DatasetInfo datasetInfo = DatasetInfo.newBuilder(datasetId)
+        .setLocation(config.get(BigQueryConsts.CONFIG_DATASET_LOCATION).asText()).build();
+    try {
+      return bigquery.create(datasetInfo);
+    } catch(Exception ex) {
+      if(ex.getMessage().indexOf("Already Exists") > -1) {
+        return bigquery.getDataset(datasetId);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Initialized bigQuery instance that will be used for verifying results of test operations
+   * and for cleaning up BigQuery dataset after the test
+   * @param config
+   * @param projectId
+   * @return
+   * @throws IOException
+   */
+  public static BigQuery initBigQuery(JsonNode config, String projectId) throws IOException {
+    final GoogleCredentials credentials = BigQueryDestination.getServiceAccountCredentials(config);
+    return BigQueryOptions.newBuilder()
+        .setProjectId(projectId)
+        .setCredentials(credentials)
+        .build()
+        .getService();
+  }
+
+  /**
+   * Deletes bigquery data set created during the test
+   * @param bigquery
+   * @param dataset
+   * @param LOGGER
+   */
+  public static void tearDownBigQuery(BigQuery bigquery, Dataset dataset, Logger LOGGER) {
+    // allows deletion of a dataset that has contents
+    final BigQuery.DatasetDeleteOption option = BigQuery.DatasetDeleteOption.deleteContents();
+    if(bigquery == null || dataset == null) {
+      return;
+    }
+    try {
+      final boolean success = bigquery.delete(dataset.getDatasetId(), option);
+      if (success) {
+        LOGGER.info("BQ Dataset " + dataset + " deleted...");
+      } else {
+        LOGGER.info("BQ Dataset cleanup for " + dataset + " failed!");
+      }
+    } catch (Exception ex) {
+      LOGGER.error("Failed to remove BigQuery resources after the test", ex);
+    }
+  }
+
+  /**
+   * Remove all the GCS output from the tests.
+   */
+  public static void tearDownGcs(AmazonS3 s3Client, JsonNode config, Logger LOGGER) {
+    if(s3Client == null) {
+      return;
+    }
+    if(BigQueryUtils.getLoadingMethod(config) != UploadingMethod.GCS) {
```

Comment:
This `if` check can probably be grouped together with the preceding `if` statement. Also just to make sure this is clear, is the reason for these empty `return` basically to say we shouldn't be logging any warnings? From what I can understand, it seems that if these values don't exist then it will log a warning in the `try-catch` block